lib/galaxy/managers/markdown_util.py

```diff
@@ -92,46 +92,35 @@ def _remap(container, line):
                 object_id = int(id_match.group(2))
                 encoded_id = trans.security.encode_id(object_id)
                 line = line.replace(id_match.group(), "{}={}".format(id_match.group(1), encoded_id))
-
             if container == "history_dataset_display":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_display(line, hda)
             elif container == "history_dataset_link":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_display(line, hda)
             elif container == "history_dataset_index":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_display(line, hda)
             elif container == "history_dataset_embedded":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_embedded(line, hda)
             elif container == "history_dataset_as_image":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_as_image(line, hda)
             elif container == "history_dataset_peek":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_peek(line, hda)
             elif container == "history_dataset_info":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_info(line, hda)
             elif container == "history_dataset_type":
-                assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_type(line, hda)
             elif container == "history_dataset_name":
-                assert object_id is not None
```

Comment:
I liked these assertions - was this needed?

Reply:
I intend to raise errors instead if that's ok. Have not pushed the commits to this PR yet tho.

Reply:
Errors are now raised and parsed in the response dictionary instead of the asserts. Additional validation could be added in follow-ups to the `validate_galaxy_markdown` in `markdown_parse.py`.

Reply:
Awesome - thanks!